### DISCUSSION TITLE
irrigation_method=3 Not Working

### DISCUSSION THREAD

**Discussion by LubinHan-Geo** [aquacropos/aquacrop#160] [General]
Hello everyone,

I recently started using AquaCrop. I try to explore the impact of different irrigation dates and depths on crop yield. According to the [guidelines](https://aquacropos.github.io/aquacrop/notebooks/AquaCrop_OSPy_Notebook_2/), I set _irrigate_schedule = IrrigationManagement(irrigation_method=3, schedule=irrigation_df)_, where _irrigation_df_ is a dataframe I created based on the requirements. I found that the yield outputs in AquaCrop with irrigation_method=3 is the same as with irrigation_method=0 (no irrigation), and the seasonal irrigation amount is zero. I noticed the same issue in the guidelines, as shown in the figure below.
![AquaCrop_OSPy_Notebook_2](https://github.com/aquacropos/aquacrop/assets/61587746/7d708736-4a70-48e8-ae88-e8665d506797)
 I am curious about such outputs because it means the specified irrigation dates and depths were not effective.

I have checked the code in irrigationManagement.py and read_irrigation_management.py. I found that in line 62 of irrigationManagement.py, an empty dataframe is imported instead of the actual irrigation file into _self.Schedule_. 
`if irrigation_method == 3:
    # wants a pandas dataframe with Date and Depth, pd.Datetime and float
    """
    dates = pd.DatetimeIndex(['20/10/1979','20/11/1979','20/12/1979'])
    depths = [25,25,25]
    irr=pd.DataFrame([dates,depths]).T
    irr.columns=['Date','Depth']
    """
    self.Schedule = pd.DataFrame(columns=["Date", "Depth"])`

I wonder if this is the reason why the irrigation amount is zero. If you know what might be causing these issues, please let me know. Thank you.

**Comment by LubinHan-Geo** [comment]
I have solved this issue. The problem is in _irrigate_schedule = IrrigationManagement(irrigation_method=3, Schedule=irrigation_schedule_df)_, what needs to be written is ’Schedule’ instead of ‘schedule‘. I found this in test_irrigation.py in the aquacrop code group in github. The writing in the guideline is wrong.
![HowToSolveIrrigationMethodEqua3](https://github.com/aquacropos/aquacrop/assets/61587746/4fd1c3a1-6a94-4247-9a6e-f53e1b913b8c)

**Comment by tfoster88** [comment]
Thanks @LubinHan-Geo for spotting this. We've got a few other updates to make in the coming weeks, so will bundle this fix in with it.
